packages/cli/src/generate.ts

```diff
@@ -28,23 +28,31 @@ import { codegen } from '@graphql-codegen/core';
 import * as typescriptPlugin from '@graphql-codegen/typescript';
 import { printSchemaWithDirectives } from '@graphql-tools/utils';
 
-import { defaultConfig, gqlessConfigPromise } from './config';
+import { defaultConfig, DUMMY_ENDPOINT, gqlessConfigPromise } from './config';
 import { formatPrettier } from './prettier';
 
 export interface GenerateOptions {
   /**
-   * Target Query Fetcher endpoint
-   * @default '/graphql'
+   * The endpoint to use for the `queryFetcher` function
    */
   endpoint?: string;
   /**
-   * Add a custom string at the beginning of the schema file, for example, add imports.
+   * Customize the TypeScript types for scalars.
+   *
+   * You can use the `preImport` option to import / define custom types.
+   *
+   * @example
+   * ```json
+   * scalarTypes: {
+   *   DateTime: "string",
+   * }
+   * ```
    */
-  preImport?: string;
+  scalarTypes?: Record<string, string>;
```

Comment:
This type name change is a breaking change and the previous version should be also provided with the `@deprecated` comment